### PR TITLE
Refactor for text encryption updates

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/4178

Also fix some tests

### CODE REVIEW

File: src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import org.craftercms.studio.api.v1.to.VersionTO;
 import org.craftercms.studio.api.v2.repository.blob.StudioBlobStore;
 import org.craftercms.studio.api.v2.repository.blob.StudioBlobStoreResolver;
-import org.craftercms.studio.impl.v2.repository.GitContentRepository;
+import org.craftercms.studio.impl.v1.repository.git.GitContentRepository;
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java:28]: Why?

File: src/test/resources/crafter/studio/extension/studio-config-override.yaml [hunk 2]
```diff
@@ -3,4 +3,4 @@
 studio.db.port: 33307
 studio.db.socket: /tmp/MariaDB4j.33307.sock
 studio.db.url: jdbc:mariadb://127.0.0.1:33307/crafter?user=crafter&password=crafter
-studio.db.initializer.url: jdbc:mariadb://127.0.0.1:33307?user=root&password=
+studio.db.initializer.url: jdbc:mariadb://127.0.0.1:33307?user=root&password=${env:MARIADB_ROOT_PASSWD}
```
Comment (sumerjabri) [inline src/test/resources/crafter/studio/extension/studio-config-override.yaml:6]: Was this missing before and causing the tests to fail?

File: pom.xml [hunk 3]
```diff
@@ -920,6 +920,7 @@
                     </systemPropertyVariables>
                     <environmentVariables>
                         <CRAFTER_DATA_DIR>target/test-resources/data</CRAFTER_DATA_DIR>
+                        <MARIADB_ROOT_PASSWD>root</MARIADB_ROOT_PASSWD>
```
Comment (sumerjabri) [inline pom.xml:923]: Was this missing before and causing problems?
Also, doesn't Studio now automatically change the password if it's root?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java (hunk 1 above)
Comment (jdrossl) [reply src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java:28]: This was caused by a bad merge and the test was not compiling

File: src/test/resources/crafter/studio/extension/studio-config-override.yaml (hunk 2 above)
Comment (jdrossl) [reply src/test/resources/crafter/studio/extension/studio-config-override.yaml:6]: yeah

File: pom.xml (hunk 3 above)
Comment (jdrossl) [reply pom.xml:923]: The default configuration is to use the env variable so for the test it has to be provided in the config, I don't know if it is changed later but it was preventing the test from running

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**